x-pack/legacy/plugins/code/server/search/repository_object_client.ts

```diff
@@ -54,6 +55,34 @@ export class RepositoryObjectClient {
     return await this.getRepositoryObject(repoUri, RepositoryRandomPathReservedField);
   }
 
+  public async getRepositories(uris: string[]): Promise<Repository[]> {
```

Comment:
nit: more like a refactor comment. This `getRepositories` is a bit duplicated with `getAllRepositories`. Can we add a 
```
private getRepositoriesInternal(index: string) {
  const res = await this.esClient.search({
      index,
      body: 
      ...
    });
}
```

to wrap the shared code together?

Reply:
Fixed